### ISSUE TITLE
It seems that the implementation of Channel only self attention and Spatial only self attention change each other.

### ISSUE THREAD

**Issue by developer0hye** [DeLightCMU/PSA#5]
Thanks for sharing your work!

https://github.com/DeLightCMU/PSA/blob/588b370d9f240d38832061a70c275cb6eb81232e/semantic-segmentation/network/PSA.py#L64-L95

![image](https://user-images.githubusercontent.com/35001605/127788727-855881b6-ad05-4829-94fa-75a8d1790acb.png)

It seems that spatial_pool function is the same with Channel-only self attention module.

**Comment by khoshsirat** [comment]
Can you explain more what do you mean?

**Comment by developer0hye** [comment]
@khoshsirat 

Does spatial pool function mean Channel-only Self-Attention?

Does channel pool function mean Spatail-only Self-Attention?

**Comment by khoshsirat** [comment]
OK, I see it now:
The `spatial_pool` function should be renamed to `channel_pool` and the `channel_pool` function should be renamed to `spatial_pool`.

I have found another discrepancy too:
In the `channel_pool` function (which should be renamed to `spatial_pool`), `softmax` is called after `matmul`. But in the paper, in the Spatial-only Self-attention block, `softmax` is used before `matmul`.

**Comment by by-seong-me** [comment]
@khoshsirat 
You are right
The location of softmax operation in channel_pool function is different with paper explanation.
What's going on?
Which one is correct?

**Comment by chunchet-ng** [comment]
Hi guys, I have created [a gist](https://gist.github.com/chunchet-ng/2292b6f9cb0954b503b957f82bbf4a04) to compare [this implementation](https://github.com/DeLightCMU/PSA/blob/20e307ad708ae6a39239aa50ffa32e0c36c92894/semantic-segmentation/network/PSA.py#L29) against [External-Attention-pytorch](https://github.com/xmu-xiaoma666/External-Attention-pytorch/blob/bf52412da8bb44d28346d1f0fc43bb3108082bf2/model/attention/PolarizedSelfAttention.py#L8)'s. Through the simple test case, I found that the outputs are different with kaiming init. 

Any idea why?
